python/grass/script/core.py

```diff
@@ -2040,15 +2040,14 @@ def _create_location_xy(database, location):
     :raises ~grass.exceptions.ScriptError:
         Raise :py:exc:`~grass.exceptions.ScriptError` on error.
     """
-    cur_dir = Path.cwd()
     try:
-        os.chdir(database)
-        permanent_dir = Path(location, "PERMANENT")
+        base_path = Path(database)
+        project_dir = base_path / location
+        permanent_dir = project_dir / "PERMANENT"
```

Comment:
```suggestion
        permanent_dir = Path(database) / location / "PERMANENT"
```

Or 
```suggestion
        permanent_dir = Path(database, location, "PERMANENT")
```

When combined with `parents=True` of mkdir

Reply:
We don’t want to fix missing folders from Path(database), only from the two others. Checking for the existence of the first path isn’t really advantageous.